doc/mainstay.md

```diff
@@ -0,0 +1,127 @@
+# Mainstay integration
+
+Options for attesting state to a mainstay proof-of-publication service. 
+
+Assumptions: 
+
+Mainstay service is available over http interface (or via SOCKS5 Tor proxy). 
+Mainstay service is available and funded with a valid `token_id` for verifiation. 
+Funding (via LN payment) is performed in advance and out of band for subscrption. (i.e. `token_id` is already performed.)
+Mainstay proofs are stored and made available, but that verification against `bitcoind` and staychain occur separately. 
```

Comment:
Okay, the mainstay mode is quite simple and I think it fits well in the civkit service framework.

There is just a relay (i.e `civkitd`) added as an intermediary between the user and mainstay service API. Multiple relays can be used to front-load or duplicate proof storages.

Good to have proofs that can be queried from service or in by the client (in case of service unavailability).

Mainstay proof format is simple, that’s good.